iOS/Flows/Conversation/Cells/EmotionCircleCollectionViewLayout.swift

```diff
@@ -0,0 +1,187 @@
+//
+//  EmotionCircleCollectionViewLayout.swift
+//  Jibber
+//
+//  Created by Martin Young on 4/6/22.
+//  Copyright © 2022 Benjamin Dodgson. All rights reserved.
+//
+
+import Foundation
+
+protocol EmotionCircleCollectionViewLayoutDataSource: AnyObject {
+    func getId(forItemAt indexPath: IndexPath) -> String
+}
+
+class EmotionCircleAttributes: UICollectionViewLayoutAttributes {
+
+    var id = String()
```

Comment:
Custom layout attributes subclass so they store their id. This allows me to keep track of which attributes correspond to which emotion when things are deleted.